axlearn/open_api/openai.py

```diff
@@ -69,20 +65,22 @@ async def async_generate(
         """
         cfg: OpenAIClient.Config = self.config
         client: AsyncOpenAI = self._client
-        assert prompt is not None or messages is not None, ValidationError(
+        prompt = request.get("prompt", None)
+        messages = request.get("messages", None)
+        assert prompt is None or messages is None, ValidationError(
```

Comment:
Is this correct? Previously we require one to be non-None, but now we check that one of them is None.

Reply:
changed. and added unit test